packages/react/src/components/form/initial.tsx

```diff
@@ -139,10 +141,18 @@ const HandleForm: React.FC<HandleFormProps> = ({
           placeholder={text["initial.handle.phone.placeholder"]}
           value={values[handleType] ?? ""}
           flag={flag}
-          onChange={registerField(handleType, (value) => {
-            if (!isValidPhoneNumber(value)) {
-              return { message: text["validationError.phoneNumber"] };
-            }
+          onChange={registerField(handleType, {
+            defaultValue:
+              shouldStoreLastID &&
```

Comment:
This is quite a complex logical expression. How do you feel about hiding this behind a dedicated function exposed from `useLastIdentifier`? Something like:
```
const { findPreviousHandle } = useLastIdentifier()

// line 145
defaultValue: findPreviousHandle(handleType, defaultValue?)
```

Then we can write tests for this and make sure it behaves exactly as we want it to